File: AndroidBasics/notes/src/withGms/java/pan/alexander/notes/di/ApplicationComponent.java
package pan.alexander.notes.di;

import javax.inject.Singleton;

import dagger.Component;
import dagger.Lazy;
import pan.alexander.notes.data.database.NotesDao;
import pan.alexander.notes.data.database.TrashDao;
import pan.alexander.notes.domain.AccountInteractor;
import pan.alexander.notes.domain.MainInteractor;

@Singleton
@Component(modules = {RepositoryModule.class, RoomModule.class, AccountModule.class})
public interface ApplicationComponent {

    Lazy<NotesDao> getNotesDao();

    Lazy<TrashDao> getTrashDao();

    Lazy<MainInteractor> getMainInteractor();

    Lazy<AccountInteractor> getAccountInteractor();
}

File: Filmrevealer/app/src/main/java/pan/alexander/filmrevealer/di/ApplicationComponent.kt
package pan.alexander.filmrevealer.di

import dagger.Component
import pan.alexander.filmrevealer.domain.interactors.MainInteractor
import pan.alexander.filmrevealer.services.RateFilmIntentService
import javax.inject.Singleton

@Singleton
@Component(
    modules = [RetrofitModule::class, RoomModule::class,
        DataStoreModule::class, DataSourcesModule::class,
        RepositoryModule::class, ManagersModule::class,
        InteractorsModule::class]
)
interface ApplicationComponent {
    fun getMainInteractor(): dagger.Lazy<MainInteractor>
    fun inject(service: RateFilmIntentService)
}
